File: Backend/Models/Entities/Delegation_Of_Authority.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ProcionAPI.Models.Entities
{
    public class Delegation_Of_Authority
    {
        [Key]
        public int Delegation_ID { get; set; }

        [Required]
        public int User_Id { get; set; }

        [Required]
        public int Admin_ID { get; set; }

        [ForeignKey("User_Id")]
        public User User { get; set; }

        [ForeignKey("Admin_ID")]
        public Admin Admin { get; set; }

        public DateTime From_Date { get; set; }
        public DateTime To_Date { get; set; }
        public string Delegation_Document { get; set; }
        public ICollection<Temporary_Access> Temporary_Access { get; set; }
    }
}

File: Frontend/src/app/Shared/DelegationOfAuthority.ts
import { Admin } from "./Admin";
import { DelegationStatus } from "./DelegationStatus"
import { User } from "./User";

export interface Delegation_Of_Authority {
  delegation_ID: Number;
  user_Id: Number;
  admin_ID: Number;
  delegationStatus_ID: Number;
  from_Date: Date;
  to_Date: Date;
  delegation_Document: String;
  delegatingParty: String;
  user: User;
  admin: Admin;
  delegation_Status: DelegationStatus;
}
